source/features/view-markdown-source.tsx

```diff
@@ -94,6 +94,10 @@ async function init(): Promise<false | void> {
 	// Add support for permalinks to the code
 	if (location.hash.startsWith('#L')) {
 		showSource();
+		// Enable highlighting
+		const originalHash = location.hash;
+		location.hash = '';
+		location.hash = originalHash;
```

Comment:
This will cause 2 history changes (i.e. to go back in history, you'll have to press `back` 2 times without visual change). Any other way to trigger `:target`?

At the very least, line 99 should use `replaceState` to remove the hash without creating an entry, so it only causes 1 history change.

Reply:
Good point! Taking a look.